Support multiple RAID controllers for compute role

mpardhi23 opened the pull request:
Hey @rpioso,
This change is to add support of multiple controller for night's watch server. 
I have implemented code for support of multiple controller in compute node which is currently tested against the available controller combination i.e. BOSS + PERC H740 . Able to create RAID1 and RAID10 correspondingly when BOSS + PERC controller found.
Testing for other scenarios is pending including deployment run. Currently I am testing this new changes and also run regression testing as well.
Note: I will enhance the code for new implementation as well, once I complete all test functionalities.

heydog15 committed:
Merge pull request #400 from gaelrehault/master

 updated default pxe nic's

gaelrehault committed:
updated sample configuration to reflect RA nic config

heydog15 committed:
Merge pull request #402 from gaelrehault/master

updated sample configuration to reflect RA nic config

gaelrehault committed:
fixed issue preventing luck version lock to properly apply to cephmetrics-ansible

heydog15 committed:
Merge pull request #405 from gaelrehault/master

fixed issue preventing yum version lock to properly apply ...

zxa13 committed:
DVR Sanity bug fix (#406)

gaelrehault committed:
clean up resource

Commit:
Fixed unity version lock bug

zxa13 committed:
DVR 6 port bug fix (#408)

heydog15 committed:
Merge pull request #409 from rajinir/unity_lock_fix2

Fixed unity version lock bug

Commit:
Unity Cinder Container: Satellite Support

heydog15 committed:
Merge pull request #407 from gaelrehault/master

+ clean up resource

gaelrehault committed:
13.2 sample switch configs

heydog15 committed:
Merge pull request #411 from gaelrehault/switch_conf_12.2

13.2 sample switch configs

heydog15 committed:
Merge pull request #410 from rajinir/unity_satelite

Unity Cinder Container: Satellite Support

gaelrehault committed:
.fix for satellited locked deployments

heydog15 committed:
Merge pull request #412 from gaelrehault/sat_lock_fix

.fix for satellited locked deployments

gaelrehault committed:
z8 lock update

heydog15 committed:
Merge pull request #414 from gaelrehault/sat_lock_fix

z8 lock update (13.2)

heydog15 committed:
Merge pull request #415 from gaelrehault/sat_lock_fix

z8 lock update (master)

gaelrehault committed:
updated version in README

gaelrehault committed:
updated sample .properties to reflect default 4 port config

gaelrehault committed:
Merge branch 'master' of https://github.com/gaelrehault/JetPack-1

gaelrehault committed:
ammended README

gaelrehault committed:
.

gaelrehault committed:
.

heydog15 committed:
Merge pull request #420 from gaelrehault/master

Updated readme  & sample (13.2)

gaelrehault committed:
up'ed virtual image size to allow fo larger rhel8 iso

heydog15 committed:
Merge pull request #426 from gaelrehault/OSP15

up'ed virtual image size to allow fo larger rhel8 iso

Commit:
Updated to python36 in kickstart

Commit:
Replaced the ntp configurations with chrony in kickstart

Commit:
firewall changes in kickstart

Commit:
Updated packages in kickstart

Commit:
chronyd is included in standard package, removed disable-NetworkManager/chronyd commands
fixed repos

heydog15 committed:
Merge pull request #432 from ahmadfsbd/OSP15

Kickstart Changes for RHEL8

ahmadfsbd committed:
Included SAH-Network Configurations through Kickstart's Network Commands (#434)

ahmadfsbd committed:
KickStart updates for SAH (#439)

zxa13 committed:
Python 3 changes for RHOSP 15 (#440)

ahmadfsbd committed:
Resolved sah network for rhel8.1, removed ssh banners (#442)

zxa13 committed:
Pyhton 3 compile error fix (#445)

ahmadfsbd committed:
Changes for creating rhel 8 based Director VM  (#449)

* Resolved sah network for rhel8.1, removed ssh banners

* Changes for creating director vm based on rhel8.1

* Director VM deployment changes for rhel8

zxa13 committed:
T10 PI format patch (#450)

ahmadfsbd committed:
Deploy OSP15 undercloud (#454)

ahmadfsbd committed:
Undercloud deployment changes for RHOSP15 Part2 (#456)

ahmadfsbd committed:
Changes for configuring container-prepare file and repo updates in ini file (#457)

Commit:
OSP16

Commit:
Merge pull request #459 from gaelrehault/OSP16

OSP16

Commit:
fixes for undercloud deployment and virt-customize

heydog15 committed:
Merge pull request #461 from ahmadfsbd/OSP16

fixes for rhosp16 undercloud deployment and virt-customize

Commit:
Update checkpoints.py

Commit:
CES-10985 - python3 edit

Commit:
CES-10988 added ironic python dependencies

Commit:
p3

cdearborn committed:
Merge pull request #465 from gaelrehault/CES-10985

CES-10985 - python3 edit

gaelrehault committed:
Update deployer.py

Commit:
CES-10989 updated ironic helper

Commit:
CES-10990 - update prep_overcloud_nodes

heydog15 committed:
Merge pull request #468 from gaelrehault/CES-10989

CES-10989 updated ironic helper

heydog15 committed:
Merge pull request #469 from gaelrehault/CES-10990

CES-10990 - update prep_overcloud_nodes

heydog15 committed:
Merge pull request #467 from gaelrehault/CES-10988

CES-10988

ahmadfsbd committed:
Code update: Preparing overcloud installation (#472)

zxa13 committed:
Added Nights Watch model to dell_systems.json file (#473)

Commit:
fix auth. errors

Commit:
.

heydog15 committed:
Merge pull request #474 from gaelrehault/OSP16

fix auth. errors

Commit:
python3 update

Commit:
miscs update for OSP16/python3

Commit:
.

heydog15 committed:
Merge pull request #476 from gaelrehault/OSP16

Osp16 miscs

Commit:
reverting bluestore workaround

Commit:
removing dashbaord

Commit:
CES-11019

Commit:
.

zxa13 committed:
CES-10736 bug fix (#487)

config_idrac.py:config_idrac has wrong model_properties default argument. Parsing through utils.py to parse JSON.

gaelrehault committed:
CES-11040 - CES-11019 (#486)

* removing dashbaord

* CES-11019

* .

ahmadfsbd committed:
RHOSP16 repo update, overcloud deploy command update (#490)

zxa13 committed:
CES-11054 bug fix (#491)

Hardcoded 'osp_admin' user in undercloud.conf changed to generic user

gaelrehault committed:
Merge branch 'OSP16' of https://github.com/gaelrehault/JetPack-1 into OSP16

heydog15 committed:
Merge pull request #492 from gaelrehault/OSP16

removed bluestore workaround

ahmadfsbd committed:
Registry update (#496)

gaelrehault committed:
updated loging.conf to reflect new PYTHONPATH on sah node

heydog15 committed:
Merge pull request #497 from gaelrehault/OSP16

updated loging.conf to reflect new PYTHONPATH on sah node

gaelrehault committed:
updated ceph repo

gaelrehault committed:
.

heydog15 committed:
Merge pull request #498 from gaelrehault/OSP16

updated ceph-ansible repo

Commit:
Templates for overcloud

Commit:
Templates for overcloud

Commit:
Resolved raid config float value

Commit:
Updated roles_data.yaml and assign_role.py

Commit:
OSP16 updates

ahmadfsbd committed:
Osp16 development (#529)

* Template changes for OSP16, along with 5-port nic

* Updated director.py, deployer.py, install-director.sh and deploy-director-vm.sh

* Various Fixes

* Raid floating value fix,undercloud ntp typo

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

Commit:
Merge remote-tracking branch 'upstream/OSP16' into OSP16

gaelrehault committed:
Osp16 (#530)

* Templates for overcloud

* Templates for overcloud

* Updated roles_data.yaml and assign_role.py

* OSP16 updates

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>
Co-authored-by: root <root@sah.OSS.LABS>

ahmadfsbd committed:
NTP pool fix (#533)

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

ahmadfsbd committed:
Minor Z2/template updates (#535)

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

Commit:
rhel8 update to by-path's

Commit:
Merge branch 'OSP16' of https://github.com/gaelrehault/JetPack-1 into OSP16

gaelrehault committed:
Osp16 (#536)

* Templates for overcloud

* Templates for overcloud

* Updated roles_data.yaml and assign_role.py

* OSP16 updates

* rhel8 update to by-path's

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>
Co-authored-by: root <root@sah.OSS.LABS>

Commit:
ssh conf update

heydog15 committed:
Merge pull request #537 from gaelrehault/OSP16

update ssh conf

dpaterson committed:
Fix utf-8 encoding issue

heydog15 committed:
Merge pull request #539 from dpaterson/patch_director_utf8

Patch director utf8 issue

Commit:
re enabling SAH as chrony source for undercloud/overcloud

cdearborn committed:
Remove T10PI workaround

This patch removes the workaround for the T10PI issue since the
permanent fix has been released and verified in the HBA330 firmware
version 16.17.01.00.

Note that JetStream and Ironic users will need to upgrade the firmware
on their HBA330s to ensure that they do not encounter the T10PI issue
once this patch is merged.

invalid-email-address committed:
re enable NTP - sah node

dpaterson committed:
Refactored to get setup_usb_idrac.py working with RHEL8/Python 3.8 (#542)

Co-authored-by: root <root@sah.r62.edge.lab>

ahmadfsbd committed:
Post-deployment data collection, & 4-port nic templates (#538)

* Post-deployment data collection & sanity

* 4-port nic templates

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

gaelrehault committed:
Update deploy-director-vm.sh

heydog15 committed:
Merge pull request #544 from gaelrehault/OSP16

re enable sah as NTP source

invalid-email-address committed:
.typo

invalid-email-address committed:
Merge branch 'OSP16' of https://github.com/gaelrehault/JetPack-1 into OSP16

dpaterson committed:
Merge pull request #545 from gaelrehault/OSP16

typo.

Commit:
Ceph Dashboard

Commit:
.dashboard

Commit:
.

Commit:
.

Commit:
.

Commit:
.

Commit:
.t

Commit:
.

heydog15 committed:
Merge pull request #546 from gaelrehault/OSP_16_dashboard

ceph dashboard

Commit:
update identify_nodes

Commit:
.

heydog15 committed:
Merge pull request #547 from gaelrehault/OSP16

fixed error appearing at the end of overcloud deployment logs

Commit:
.removed docs

Commit:
header year update

heydog15 committed:
Merge pull request #548 from gaelrehault/remove_docs

Removde docs

heydog15 committed:
Merge pull request #549 from gaelrehault/OSP16

headers year update

heydog15 committed:
Merge pull request #541 from cdearborn/remove_t10pi_workaround

Remove T10PI workaround

gaelrehault committed:
Up'ed reset idrac wait time

heydog15 committed:
Merge pull request #550 from gaelrehault/OSP16

Up'ed reset idrac wait time

dpaterson committed:
Add edge subnet support for undercloud (#551)

* Add edge subnet support for undercloud

* Add edge subnet support for undercloud

* fix method name, missing _

* Remove Dashboard references

* Fix _init_

* remove dashboard from sample inis

* Refactor config

* Add nic configs directory constant

* Add missing _generate methods needed for edge subnets

* Add missing funtions

* Add dashboard params back in

* Add newline to end of file

* cntlplane-subnet gateway is supposed to be director provisioning_ip not ini.provisioning_gateway!

Co-authored-by: David Paterson <davpat2112@yahoo.com>

Commit:
nvme drives support

ahmadfsbd committed:
Osp16 outdated perf settings (#552)

* Rmoved outdated perf settings

* Extra

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

ahmadfsbd committed:
Reviewed and updated extra-configs in dell-environment.yaml (#556)

Co-authored-by: root <root@r7-sah.r7.cluster>

ahmadfsbd committed:
Reviewed performance tunings in first-boot.yaml (#560)

Co-authored-by: root <root@r7-sah.r7.cluster>

heydog15 committed:
Merge pull request #554 from gaelrehault/OSP16

nvme drives support

gaelrehault committed:
HCI support (#559)

Co-authored-by: root <root@sah.OSS.LABS>

Commit:
updated path to ipxe image

ahmadfsbd committed:
HCI Network Templates (#562)

Co-authored-by: root <root@r7-sah.r7.cluster>

heydog15 committed:
Merge pull request #563 from gaelrehault/OSP16

updated path to inspector image

invalid-email-address committed:
Backport PowerMax and Unity into JS-16.1/OSP16 branch

heydog15 committed:
Merge pull request #564 from rajinir/OSP16

Backport PowerMax and Unity into JS-16.1/OSP16 branch

invalid-email-address committed:
Fixing the idendation

heydog15 committed:
Merge pull request #565 from rajinir/OSP16

Fixing the idendation

invalid-email-address committed:
Fixing comments

heydog15 committed:
Merge pull request #566 from rajinir/OSP16

Fixing comments

ahmadfsbd committed:
HCI support fixes/updates (#568)

Co-authored-by: Muhammad Ahmad <root@r7-sah.r7.cluster>

Commit:
Updated overcloud deployment status check

heydog15 committed:
Merge pull request #569 from gaelrehault/OSP16

Updated overcloud deployment status check

dpaterson committed:
Overcloud edge support (#567)

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* update is_coherent

* remove bogus file and duplicate arg in deploy-overcloud.py

* Change message to grep for

* Fixed cntlplane-subnet, broke during rebase

* Remove template ceph-osd-config.yaml from overcloud deploy command as it is not used in OSP16

* Getting rid of extraneous jinja templates for now, also updating ExternalInterfaceDefaultRoute in 5 port controller template

* Move network-isolation and network-environment templates to end of deployement args

* Removing dell-enironment jinja template until needed

* Remove duplicate jinja2 constant

* Remove defaults for consistancy

* Add jinja2 package to kickstart

* Remove overlapping templates where jinja is used, also refactored some function names

* Migrate nic_environment.yaml generation to jinja

* Remove controller yaml as it is generated via Jinja

* Small edits, removing old comments and duplicate lines

* Update copyright, delete unused imports...

* Update description in templates

* Remove roles_data.yaml as it is now rendered via jinja

* Remove duplicate network-data argument

* Add private_api_gateway config

* Refactored template generation so that only edge sites are generated

* Refactored template generation so that only edge sites are generated

* Remove duplicate computehci_nodes

* Removing deplrecated nova scheduler filters for now until we have workaround for fixing missing filter classes

Co-authored-by: David Paterson <davpat2112@yahoo.com>

ahmadfsbd committed:
Bits for 16.1 deployment with rhel8.2 (#571)

Co-authored-by: root <root@sah.oss.labs>

Commit:
updated samples 's repos

Commit:
Numa code updated, added missing parameters to sample ini files

Commit:
Adding OVS support to OSP16.1

Commit:
Adding OVS support to OSP16.1

Commit:
Merge branch 'enable_ovs' of https://github.com/kalagasp/JetPack into enable_ovs

Commit:
tab to space

dpaterson committed:
Merge pull request #576 from kalagasp/enable_ovs

Enable ovs

Commit:
added a patch to fix issue with nova scheduler preventing instances creation (bz 1846020)

dpaterson committed:
Merge pull request #577 from gaelrehault/bz_1846020

CES-11250 part II

heydog15 committed:
Merge pull request #573 from gaelrehault/JS-16.1

updated samples 's repos

heydog15 committed:
Merge pull request #575 from ahmadfsbd/JS-16.1

Numa code updated, added missing parameters to sample ini files

invalid-email-address committed:
Updated Unity Container URL

heydog15 committed:
Merge pull request #579 from rajinir/JS-16.1

Updated Unity Container URL

Commit:
Updated NFV nova scheduler filters

heydog15 committed:
Merge pull request #580 from ahmadfsbd/JS-16.1

Updated NFV nova scheduler filters

ahmadfsbd committed:
Vlan networks fix, Isol Cpu Fix for numa and DPDK (#582)

Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>

Commit:
.

Commit:
.

heydog15 committed:
Merge pull request #584 from gaelrehault/satellite

Satellite for 16 .1

Commit:
ES-11554

ahmadfsbd committed:
SRIOV deployment support, templates (#583)

Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>

heydog15 committed:
Merge pull request #585 from gaelrehault/CES-11554

CES-11554

Commit:
.updated network validation

Commit:
updated register overcloud

heydog15 committed:
Merge pull request #586 from gaelrehault/JS-16.1

updated network validation & register overcloud scripts

Commit:
Updated subscription.json for RHEL onl

heydog15 committed:
Merge pull request #588 from heydog15/JS-16.1

Updated subscription.json for RHEL onl

ahmadfsbd committed:
DPDK support, Heat/Nic Templates (#589)

Co-authored-by: Muhammad Ahmad <stack@director.oss.labs>

ahmadfsbd committed:
Support/templates for OVS-DPDK-SRIOV (#590)

Co-authored-by: Muhammad Ahmad <sambhu.kalaga@dell.com>

heydog15 committed:
Updated .ini file for rhel82 and updated Satellite doc path (#591)

* Updated .ini file for rhel82 and updated Satellite doc path

* Updated .ini file for rhel82 and updated Satellite doc path

* Updated sample_hci_profile.ini to match csp and xsp

Co-authored-by: root <root@rhel8auto_kurt.oss.labs>

Commit:
Unity Custom Container Changes for 16.1 (#592)

Co-authored-by: root <root@localhost.localdomain>

cdearborn committed:
Don't use FORCE when clearing the job queue (#593)

Clearing the job queue with FORCE causes a service internal to the iDRAC
to restart.  Until this service is fully up, information queried from
the iDRAC may not be complete. This patch changes config_idrac to
not use the FORCE option to resolve related issues.

ahmadfsbd committed:
DVR Support for 16.1 (#595)

Co-authored-by: root <root@sah.oss.labs>

Commit:
pull ironic from rpm's rather than pip to ensure we get the right version

ahmadfsbd committed:
DVR template re-order (#598)

Co-authored-by: root <root@sah.oss.labs>

heydog15 committed:
Merge pull request #600 from gaelrehault/JS-16.1

pull ironic from rpm's rather than pip to ensure we get the right ver…

rpioso committed:
Merge pull request #596 from rpioso/experimental-upstream-ironic-JS-16.1-00

Merge JS-16.1 into experimental-upstream-ironic-JS-16.1-00

ahmadfsbd committed:
SRIOV-Offload support, updates and templates (#599)

Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>

rpioso committed:
CES-11261 Update Ironic undercloud patches (#601)

invalid-email-address committed:
Unity Manila Custom Container

gaelrehault committed:
CES-11261 Update Ironic undercloud patches (#601) (#602)

Co-authored-by: Richard G. Pioso <richard.pioso@dell.com>

heydog15 committed:
Merge pull request #603 from rajinir/unity_manila

Unity Manila Custom Container

Commit:
updated patch handling for OSP16

heydog15 committed:
Merge pull request #604 from gaelrehault/CES-11261

updated patch handling for OSP16

himanshusingh2 committed:
Update ironic.patch file to use iDRAC driver supported interfaces.

rpioso committed:
Resolve merge conflicts

Conflicts:
	src/pilot/ironic.patch

himanshusingh2 committed:
CES-11519 Update scripts to use the hybrid idrac driver/hardware type (#574)

cdearborn committed:
Merge pull request #605 from himanshusingh2/CES-11611

CES-11611 Configure ironic to use idrac HW type supported interfaces

gaelrehault committed:
CES-11261 updated assign_role ... (#607)

gaelrehault committed:
Merge branch 'JS-16.1' into experimental-upstream-ironic-JS-16.1-00

heydog15 committed:
Merge pull request #608 from dsp-jetpack/experimental-upstream-ironic-JS-16.1-00

Pulling recent ironic changes

Commit:
.

heydog15 committed:
Merge pull request #611 from gaelrehault/JS-16.1

Fixed EOF

Commit:
updated dns check

heydog15 committed:
Merge pull request #613 from gaelrehault/JS-16.1

updated dns check

dpaterson committed:
Deploy edge sites seperate stacks (#606)

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* update is_coherent

* remove bogus file and duplicate arg in deploy-overcloud.py

* Change message to grep for

* Fixed cntlplane-subnet, broke during rebase

* Remove template ceph-osd-config.yaml from overcloud deploy command as it is not used in OSP16

* Getting rid of extraneous jinja templates for now, also updating ExternalInterfaceDefaultRoute in 5 port controller template

* Move network-isolation and network-environment templates to end of deployement args

* Removing dell-enironment jinja template until needed

* Remove duplicate jinja2 constant

* Remove defaults for consistancy

* Add jinja2 package to kickstart

* Remove overlapping templates where jinja is used, also refactored some function names

* Migrate nic_environment.yaml generation to jinja

* Remove controller yaml as it is generated via Jinja

* Small edits, removing old comments and duplicate lines

* Update copyright, delete unused imports...

* Update description in templates

* Remove roles_data.yaml as it is now rendered via jinja

* Remove duplicate network-data argument

* Add private_api_gateway config

* Refactored template generation so that only edge sites are generated

* Refactored template generation so that only edge sites are generated

* Remove duplicate computehci_nodes

* Removing deplrecated nova scheduler filters for now until we have workaround for fixing missing filter classes

* Initial checkin for multi-stack edge deployments, WIP

* Refactoring render_and_upload_* methods to be able *_all and breaking out template generation code to handle a single edge site, This is so we can deploy all edge sites or a single edge site via cli

* Finshed refactoring template generation, now working through discover nodes, config idrac and assign role for edge sites

* Discover nodes and config idracs multi-stack edge support

* Completed import edge nodes and validating edge nodes are in ironic database

* Introspection for edge sites, partially done

* Checkpoint for introspection updated to support edge sites, introspection for edge complete

* Assign role completed for edge sites

* fix path to first-boot.yaml template and fix quotes in %hieradata template vars

* Fix provisioning interface for director vm, was enp3s0, should have been enp2s0

* Added cli for deploying edge sites

* Lots of refactoring for multistack support, WIP

* fixed public api interface for creating director vm

* fix sample .properties for interfaces, they were in sah definition instead of director def by mistake. Removed some debugging as well.

* Add deployer argument to list all edge site nodes and networking info

* refactor route creation to use CLI instead of modifying network-scripts manually.  Using NetworkManager wherever possible.  It will be additional work to get NetworkManager managing OVS bridge routes as our current implementation is not using NM to managem OVS objects at all.

* Finished all sah and director VM route/unroute refactoring, still have issue with Director VM requiring reboot post routes

* Updated provisioning routes to use network-scripts as it's the only way to make them persistent

* Refactored updating undercloud for edge sites.

* Refactor functions to only require node_type arg and derive other needed data

* Fix building container image for, have to make call as sudo user

* Single edge site deploy via automation, WIP

* Fix overcloud export target directory and filename

* Added additional edge jinja templates, fixed cpl issues with ControlPlane variables

* added _fetch_network_environment_params function and some test code

* Add function to see if edge subnet exists, also fixed convention for control plane role attributes, the MUST be <role>ControlPlane<attribute> they were alll ControlPlane<role><attribute> like the rest of the network role attributes.  ControlPlane is special case.

* DefaultBondMTU must have Mtu in camel-case DefaultBondMtu

* fix hiera calls and added delete_edge_site

* added routes for edge nodes back to control plane networks and from controllers to edge sites

* Added edge site deletion and controller routing for edge sites

* Fetch director domain interfaces from virsh

* Fixed ExtraConfig params for genteated dell-environment_[edge_site].yaml.  Edge sites and instaces deployed to edge AZ functional, yay

* Added mutli-site support for sanity test, add nova-az-cofig.yaml to overcloud site-name.yaml template

* Fix sanity for multi-stack support, update routes on sah to use device reapply instead of connection up, cleanup debugging code and commented out code

* Fix edge provisioning routes for edge sites, need to restart interface, restarting bridge is not enough

* Backout debugging code in Ssh.py

* Remove unused code, _has_edge_sites, as we explicitly deploy/tear down edge sites in 16.1

* Remove comments and debugging code

* Remove deploy/delete all edge site deployment args and functions, moved all _all code to deploy_edge_sites_sep_stacks_with_all, in case we want to revist

* Remove debugging code from sah and director_vm modules

* Remove unused, commented out import

* Fix issues with controller routes still showing up after route deletion and ifup-route, solution is to call ip route del as well, lame

* Cleanup logic for deciding when to deploy Director and Overcloud

* Tons of refactoring after logic evaluation with team, WIP

* Parially done with PR change requests, WIP

* Improve overcloud node import into ironic validation as requested in PR review

* Removing debugging code

* Fix edge site list

* Remove noisy log statement

* When re-deploying edge sites first delete the existing edge stack if exists and blow away the edge nodes in ironic database

* Fix assign role from rebase

* Fix assign role from rebase

* Fix assign role from rebase

* Fix assign role from rebase

* Refactored deploy/delete edge site a bit, when re-deploying an edge site always delete existing stack, nodes and templates for that site. Originally these resources were only deleted if the edge site stack was found.  Now each resource is deleted independently

* Need to append to undercloud.conf:subnets vs replace

* Instead of patching undercloud conf files we need to use hieradata overrides

* Move config changes from patch files to undercloud hieradata overrrides

* Move config changes from patch files to undercloud hieradata overrrides

* Move config changes from patch files to undercloud hieradata overrrides

* Move config changes from patch files to undercloud hieradata overrrides

Co-authored-by: David Paterson <davpat2112@yahoo.com>

dpaterson committed:
Update ini samples edge stanzas (#614)

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* update is_coherent

* remove bogus file and duplicate arg in deploy-overcloud.py

* Change message to grep for

* Fixed cntlplane-subnet, broke during rebase

* Remove template ceph-osd-config.yaml from overcloud deploy command as it is not used in OSP16

* Getting rid of extraneous jinja templates for now, also updating ExternalInterfaceDefaultRoute in 5 port controller template

* Move network-isolation and network-environment templates to end of deployement args

* Removing dell-enironment jinja template until needed

* Remove duplicate jinja2 constant

* Remove defaults for consistancy

* Add jinja2 package to kickstart

* Remove overlapping templates where jinja is used, also refactored some function names

* Migrate nic_environment.yaml generation to jinja

* Remove controller yaml as it is generated via Jinja

* Small edits, removing old comments and duplicate lines

* Update copyright, delete unused imports...

* Update description in templates

* Remove roles_data.yaml as it is now rendered via jinja

* Remove duplicate network-data argument

* Add private_api_gateway config

* Refactored template generation so that only edge sites are generated

* Refactored template generation so that only edge sites are generated

* Remove duplicate computehci_nodes

* Removing deplrecated nova scheduler filters for now until we have workaround for fixing missing filter classes

* Update sample inis with the latest edge site attributes

Co-authored-by: David Paterson <davpat2112@yahoo.com>

Commit:
Fixing CES-11633 Sanity does not support for Unity in JS-16.1 release (#610)

The support for sanity test for manila shares was not migrated
from JS13.3

Co-authored-by: root <root@localhost.localdomain>

Commit:
updated assign role

heydog15 committed:
Merge pull request #615 from gaelrehault/JS-16.1

updated assign role - JS-16.1

heydog15 committed:
Updated director_vlock (#621)

Co-authored-by: root <root@rhel8auto_kurt.oss.labs>

dpaterson committed:
Add 4 port edge support (#625)

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* update is_coherent

* remove bogus file and duplicate arg in deploy-overcloud.py

* Change message to grep for

* Fixed cntlplane-subnet, broke during rebase

* Remove template ceph-osd-config.yaml from overcloud deploy command as it is not used in OSP16

* Getting rid of extraneous jinja templates for now, also updating ExternalInterfaceDefaultRoute in 5 port controller template

* Move network-isolation and network-environment templates to end of deployement args

* Removing dell-enironment jinja template until needed

* Remove duplicate jinja2 constant

* Remove defaults for consistancy

* Add jinja2 package to kickstart

* Remove overlapping templates where jinja is used, also refactored some function names

* Migrate nic_environment.yaml generation to jinja

* Remove controller yaml as it is generated via Jinja

* Small edits, removing old comments and duplicate lines

* Update copyright, delete unused imports...

* Update description in templates

* Remove roles_data.yaml as it is now rendered via jinja

* Remove duplicate network-data argument

* Add private_api_gateway config

* Refactored template generation so that only edge sites are generated

* Refactored template generation so that only edge sites are generated

* Remove duplicate computehci_nodes

* Removing deplrecated nova scheduler filters for now until we have workaround for fixing missing filter classes

* Added 4-port config, WIP

* Added 4-port config, fix extrananeous endif tag, WIP

* moved nic_environment_edge_j2.yaml to parent directory as it is not port dependent

* Add idrac_card.patch for XE2420 4 port config to work

* Add idrac_card.patch for XE2420 4 port config to work

* Fix idrac_card.patch

* Fix idrac_card.patch

Co-authored-by: David Paterson <davpat2112@yahoo.com>

Commit:
.fix issue assigning root device

heydog15 committed:
Merge pull request #627 from gaelrehault/JS-16.1

CES-11670

Commit:
CES-11667 - updated sanity test

heydog15 committed:
Merge pull request #628 from gaelrehault/CES-11667

CES-11667 - updated sanity test

Commit:
container lock

heydog15 committed:
Merge pull request #631 from gaelrehault/JS-16.1

container lock

Commit:
Vlock for containers (#632)

Co-authored-by: root <root@localhost.localdomain>

Commit:
enabling version locking

heydog15 committed:
Updated sample_csp.ini (#634)

Co-authored-by: root <root@rhel8auto_kurt.oss.labs>

heydog15 committed:
Updated the other 2 .ini files (#636)

Co-authored-by: root <root@rhel8auto_kurt.oss.labs>

heydog15 committed:
Merge pull request #635 from gaelrehault/JS-16.1

enabling version locking

Commit:
fixed anaconda dns

heydog15 committed:
Merge pull request #637 from gaelrehault/JS-16.1

fixed anaconda dns

Commit:
.

heydog15 committed:
Merge pull request #639 from gaelrehault/JS-16.1

.

Commit:
fat fingers

gaelrehault committed:
fat fingers (#641)

Co-authored-by: root <root@sah.OSS.LABS>

Commit:
fixing container lock

Commit:
lock edits

dpaterson committed:
Add 8 port DPDK/SRIOV edge support (#633)

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* Edge WIP - merge master and refactoring

* Edge WIP - merge master and refactoring

* Update director mgmt interface for edge routes

* Jinja is functional on SAH

* network_data.yaml jinja2 generation done

* Initial commit of jinja2 nic templates

* 5 port compute nic template generation done

* Fix missing closed paren

* update_nic_environment_edge updated to use jinja templates

* jinja2-ized update_stamp_nic_config_routes_edge

* Jinja2-ized roles data

* Generating static ip template with jinja2\nSupports both overcloud ips and edge ips

* Generating static vip template with jinja2

* Migrated network isolation template to jinja2

* Fix sanity ini config

* update is_coherent

* remove bogus file and duplicate arg in deploy-overcloud.py

* Change message to grep for

* Fixed cntlplane-subnet, broke during rebase

* Remove template ceph-osd-config.yaml from overcloud deploy command as it is not used in OSP16

* Getting rid of extraneous jinja templates for now, also updating ExternalInterfaceDefaultRoute in 5 port controller template

* Move network-isolation and network-environment templates to end of deployement args

* Removing dell-enironment jinja template until needed

* Remove duplicate jinja2 constant

* Remove defaults for consistancy

* Add jinja2 package to kickstart

* Remove overlapping templates where jinja is used, also refactored some function names

* Migrate nic_environment.yaml generation to jinja

* Remove controller yaml as it is generated via Jinja

* Small edits, removing old comments and duplicate lines

* Update copyright, delete unused imports...

* Update description in templates

* Remove roles_data.yaml as it is now rendered via jinja

* Remove duplicate network-data argument

* Add private_api_gateway config

* Refactored template generation so that only edge sites are generated

* Refactored template generation so that only edge sites are generated

* Remove duplicate computehci_nodes

* Removing deplrecated nova scheduler filters for now until we have workaround for fixing missing filter classes

* First pass at 8 port dpdk

* Fix nested members for dpdk bond, went one level too deep

* Fix dpdk interface name consistancy

* WIP for dpdk support, partially implemented dell_nfv_edge.py

* WIP for dpdk support, partially implemented dell_nfv_edge.py

* Some refacorting to simplify if and what NFV features are enabled and fetch the required node inventory attributes needed for template generation and ship the paramters as a json blob back to the sah

* Fix typo

* Refactor dell_nfv_edge to extend dell_nfv

* First pass at implementing sriov for edge sites, WIP

* Fix 8 port sriov template generation

* Fix 8 port sriov template generation

* Add routes for vlans on 8 port sriov template generation

* Remove debugging code

* Refactor nfv_type to be fetched from _get_nfv_type() everywhere

* Clean up loggging calls used for debugging

* Add new NFV feature specific attributes to edge site definitions

* Fix OVS DPDK detection

* Removing commented out debugging code

* Refactor last place in code where _get_nfv_type() should be used.

* Always add NovaComputeCpuDedicatedSet is numa is enabled regardless of DPDK or SRIOV settings

* Add ovs_extra to DPDK compute template rendering as it's needed for tenant tunnel traffic

Co-authored-by: David Paterson <davpat2112@yahoo.com>

heydog15 committed:
Merge pull request #642 from gaelrehault/JS-16.1

Fixing locks format for z1.

mpardhi23 committed:
Added support of multiple controller for night's watch server

rpioso committed:
CES-11691 Update Ironic undercloud patches for OSP 16.1.2 (#645)

* CES-11691 Update Ironic undercloud patches for OSP 16.1.2

* Remove monkey patches for Red Hat bugz 1846020

This change removes the monkey patches which worked around Red Hat
Bugzilla bug 1846020. [1]

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1846020

gaelrehault committed:
Z2  (#643)

* .

* .

* locking off

* .

Co-authored-by: root <root@sah.OSS.LABS>
Co-authored-by: root <root@sah.r180.dsp.labs>

jproque-dell committed:
Including Powerflex work into new branch - initial PR (#646)

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* Removed RamFilter, DiskFilter and CoreFilter as they have been removed from Train scheduling host filters

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* Removed RamFilter, DiskFilter and CoreFilter as they have been removed from Train scheduling host filters

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* Js 16.1 powerflex automation support including gateway install/config (#594)

* PowerFlex 3.5 automation support

* PowerFlex 3.5 automation support

* Vlan networks fix, Isol Cpu Fix for numa and DPDK (#582)

Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>

* .

* .

* ES-11554

* SRIOV deployment support, templates (#583)

Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>

* .updated network validation

* updated register overcloud

* Updated subscription.json for RHEL onl

* Adding gateway deployment

* DPDK support, Heat/Nic Templates (#589)

Co-authored-by: Muhammad Ahmad <stack@director.oss.labs>

* Support/templates for OVS-DPDK-SRIOV (#590)

Co-authored-by: Muhammad Ahmad <sambhu.kalaga@dell.com>

* Updated .ini file for rhel82 and updated Satellite doc path (#591)

* Updated .ini file for rhel82 and updated Satellite doc path

* Updated .ini file for rhel82 and updated Satellite doc path

* Updated sample_hci_profile.ini to match csp and xsp

Co-authored-by: root <root@rhel8auto_kurt.oss.labs>

* Unity Custom Container Changes for 16.1 (#592)

Co-authored-by: root <root@localhost.localdomain>

* Don't use FORCE when clearing the job queue (#593)

Clearing the job queue with FORCE causes a service internal to the iDRAC
to restart.  Until this service is fully up, information queried from
the iDRAC may not be complete. This patch changes config_idrac to
not use the FORCE option to resolve related issues.

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* PowerFlex 3.5 automation support

* PowerFlex 3.5 automation support

* PowerFlex Settings Support (#561)

Ini file changes

Co-authored-by: root <root@localhost.localdomain>

* Removed RamFilter, DiskFilter and CoreFilter as they have been removed from Train scheduling host filters

* Adding gateway deployment

* Applying changes based upon Gael's review

* Applying changes based upon Gael's review

* Making powerflex backend name changeable as per Rajini's suggestion

* Removing host scanning to determine if we need to collect LIA and MDM certificates

* Removing config_interface

* Applying changes as per Rajini's last review

* Fixing typo

* Changing to use the env file instead of the ExtraConfig as pers Rajini's suggestion

* Cloning the tripleo-powerflex repository when deploying instead of packaging everything

* Removing testing file

* Cleaning up profiles

* Extra check to enable overcloud customization with multiple conditions

* Cleaning up .ini files

* Wiping disk if previous data was on the powerflex nodes

* Removing static build of disks

Co-authored-by: Muhammad Ahmad <43810745+ahmadfsbd@users.noreply.github.com>
Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>
Co-authored-by: heydog15 <heydog15@users.noreply.github.com>
Co-authored-by: root <root@rhel8auto_kurt.oss.labs>
Co-authored-by: Muhammad Ahmad <stack@director.oss.labs>
Co-authored-by: Muhammad Ahmad <sambhu.kalaga@dell.com>
Co-authored-by: Rajini Karthik <rajini.karthik@dell.com>
Co-authored-by: root <root@localhost.localdomain>
Co-authored-by: Chris Dearborn <Christopher.Dearborn@dell.com>

* Fixing missing SDC package on controllers (#624)

* Fixing missing SDC package on controllers

* Adding custom mapping for containers

* Adding directory creation requirement for Glance support with PowerFlex

* Adding PowerFlex presentation server support

* Adding cinder backend for Glance with PowerFlex

* Adding cinder backend for Glance with PowerFlex

* Fixing backend name

* Adding directory creation requirement for Glance support with PowerFlex and Glance backend (#629)

* Adding directory creation requirement for Glance support with PowerFlex

* Adding PowerFlex presentation server support

* Adding cinder backend for Glance with PowerFlex

* Adding cinder backend for Glance with PowerFlex

* Fixing powerflex backend name

* Adding PowerFlex support for HCI deployment

* adding support for PowerFlex Ready Nodes

* Adding bit for z2

Co-authored-by: Rajini Karthik <rajini.karthik@dell.com>
Co-authored-by: root <root@localhost.localdomain>
Co-authored-by: Muhammad Ahmad <43810745+ahmadfsbd@users.noreply.github.com>
Co-authored-by: MUHAMMAD AHMAD <root@sah.oss.labs>
Co-authored-by: heydog15 <heydog15@users.noreply.github.com>
Co-authored-by: root <root@rhel8auto_kurt.oss.labs>
Co-authored-by: Muhammad Ahmad <stack@director.oss.labs>
Co-authored-by: Muhammad Ahmad <sambhu.kalaga@dell.com>
Co-authored-by: Chris Dearborn <Christopher.Dearborn@dell.com>

Commit:
.update for non flex deployments

jproque-dell committed:
Creating functions for powerflex vms and cleaning old bits (#648)

Commit:
flex related updates

Commit:
.

Commit:
.:

jproque-dell committed:
Merge pull request #650 from gaelrehault/JS-16.1.20-11

powerflex related updates

mpardhi23 committed:
added multiple controller support on controller node

mpardhi23 commented:
Hi Guys,
I have tested multiple controller code by running OSP deployment by configuring night's watch server which is having BOSS + PERC combination as compute node in properties file . Assign_role executed successfully on all nodes and able to create RAID1 and RAID10 correspondingly on night's watch server.  Deployment is successful till overcloud installation.

rpioso commented:
@dpaterson, do you still plan to review this pull request? If not, perhaps @gaelrehault or @heydog15 could.

dpaterson commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
```
Is it cleaner to only pass in drac_client and fetch raid_controllers in this method as we do with all_physical_disks...?

mpardhi23 replied:
If we fetch raid_controllers in this method then I need to change the existing code , where this function get invoked in  define_target_raid_config function. In this function we already fetched raid_controllers.
line no. 263 : logical_disks = define_controller_logical_disks(drac_client, raid_controller_ids) 
I think we can keep it as is.

dpaterson commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
-    if raid_10_logical_disk is None:
+    boss_controller = [cntrl for cntrl in raid_controllers
+                       if drac_client.is_boss_controller(cntrl)]
+
```
Perhaps add new function _fetch_boss_controller is cleaner than this construct, also doesn't this return a array so for it to be correct it would be:
[cntrl for cntrl in raid_controller if drac_client.is_boss_controller(cntrl)][0] ?

mpardhi23 replied:
Agree. I will add new function which fetch the boss controller and return None or boss card.

mpardhi23 replied:
Done

dpaterson commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
+
+    if boss_controller:
+        if check_cntlr_physical_disks_len(
+           raid_cntlr_physical_disks[boss_controller[0]]):
```
Not a fan of boss_controller being a list, there can only be one boss card per system so we should just have single method _fetch_boss_card or something like that, that returns None or the boss card, not a list.

mpardhi23 replied:
Okay , will do this changes and push the latest changes.

mpardhi23 replied:
Done

dpaterson commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
+
+    if boss_controller:
+        if check_cntlr_physical_disks_len(
+           raid_cntlr_physical_disks[boss_controller[0]]):
```
Same comment as above, only one boss card per box is possible.

mpardhi23 replied:
sure

mpardhi23 replied:
Done

dpaterson commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,120 @@ def get_raid_controller_ids(drac_client):
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_compute_logical_disks(drac_client, raid_controllers):
+    all_physical_disks = drac_client.list_physical_disks()
```
Same comment as above, use drac_client to fetch controllers just like we fetch all_physical_disks

mpardhi23 replied:
I think we can keep it as is.

dpaterson reviewed (changes requested):
Couple of small changes would make this cleaner but I don't see anything wrong with logic at first sight,  has this been tested against real hardware?

mpardhi23 commented:
> Couple of small changes would make this cleaner but I don't see anything wrong with logic at first sight, has this been tested against real hardware?

Yes I have tested these changes against night's watch server XE2420.

mpardhi23 committed:
added function to fetch boss controller

dpaterson reviewed (approved):
Thanks for implementing suggested changes

mpardhi23 commented:
> Thanks for implementing suggested changes

Thanks David!

ajya commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
-    raid_10_logical_disk = define_single_raid_10_logical_disk(
-        drac_client, raid_controller_ids[0])
+    # Get the drives controlled by the RAID controller
+    raid_cntlr_physical_disks = {}
```
this can be shortened by using defaultdict(list)

ajya commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
+            LOG.critical("The BOSS card has only 1 SSD. "
+                         "2 SSDs are needed to configure a RAID 1")
+            return None
+    else:
```
no need for else here - RAID 10 should be executed in both cases - when there is boss controller and when there is not. The remaining code below that tries to create RAID10 again can be removed.

mpardhi23 replied:
yes removed else and remaining below code from here.

ajya commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
+    raid_cntlr_physical_disks = {}
+    for disk in all_physical_disks:
+        if disk.controller in raid_controllers:
+            if disk.controller in raid_cntlr_physical_disks:
```
this can be shortened by using defaultdict(list) for raid_cntlr_physical_disks

rpioso replied:
@mpardhi23, you may find https://stackoverflow.com/questions/26367812/appending-to-list-in-python-dictionary helpful.

mpardhi23 replied:
Done

ajya commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
+                         "2 SSDs are needed to configure a RAID 1")
+            return None
+    else:
+        os_logical_disk = define_single_raid_10_logical_disk(
```
update define_single_raid_10_logical_disk to have optional is_root_volume param with default True (as it is used now).

Then based on scenario: 
1) if there is boss controller, pass is_root_volume=False
2) is there is no boss controller, pass is_root_volume=True

mpardhi23 replied:
Done

ajya commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_compute_logical_disks(drac_client, raid_controllers):
```
this method looks very similar to define_controller_logical_disks and too much of code is duplicating, need to eliminate duplication. Two of the options:

1) merge the method into one, named e.g. define_controller_or_compute_logical_disks and update where they both are called.
2) introduce private helper method, e.g., _define_controller_or_compute_logical_disks and call it from these 2 methods. This way don't have to change all the places where these 2 are called now.

rpioso replied:
FWIW, I prefer option 2. That option makes it less likely that a programmer will introduce an error by not updating all of the callers. Since this script does not have any unit tests, such errors can very easily go undetected.

mpardhi23 replied:
Added _define_controller_or_compute_logical_disks method which defines the compute and controller logical disk.

rpioso commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
+    if boss_controller:
+        if check_cntlr_physical_disks_len(
+           raid_cntlr_physical_disks[boss_controller]):
+            os_logical_disk = define_storage_operating_system_logical_disk(
```
Having this function, which is used for the control and compute roles, call another function which is named for the storage role can be confusing.

Let us make the following function name changes:

1. `define_storage_operating_system_logical_disk` to `define_operating_system_logical_disk`
2. `find_physical_disks_for_storage_os` to `find_physical_disks_for_os`

mpardhi23 replied:
Done

rpioso reviewed (changes requested):
Thank you, @mpardhi23! I have left several in-line comments which offer suggestions for improvement.

mpardhi23 committed:
added define_controller_or_compute_logical_disk method

mpardhi23 commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+def define_controller_logical_disks(drac_client, raid_controllers):
```
Thanks Richard for review comments, replaced raid_controllers to raid_controller_ids. as its listing only controller ids.

mpardhi23 commented:
Hi Richard and Aija,
I have addressed all the review comments you mentioned. And also tested those changes with the below controllers available on server.
1. H740P 
2. BOSS + H740P
Need to test it against another controllers which are available on another servers (different stamps)

ajya commented on src/pilot/assign_role.py:
```diff
@@ -297,52 +297,52 @@ def get_raid_controller_ids(drac_client):
-            "Found more than one RAID controller:\n  {}".format(
-                "\n  ".join(raid_controller_ids)))
-        return None
+    is_root_volume=True
```
let's use `boss_controller` to track if RAID 10 will be root volume or not. This way can avoid introducing another variable. After removing `is_root_volume`, on line 335 use `not boss_controller`.

mpardhi23 replied:
removed _is_root_volume_ and used not _boss_controller_.

ajya commented on src/pilot/assign_role.py:
```diff
@@ -297,52 +297,52 @@ def get_raid_controller_ids(drac_client):
-    logical_disks = list()
-
-    # Add the disk to the list only if it is not empty.
     if raid_10_logical_disk:
```
if there is no RAID 10 created, let's return `None` as it is mandatory to have it on compute and controller nodes.

mpardhi23 replied:
Done

ajya commented on src/pilot/assign_role.py:
```diff
@@ -405,6 +405,13 @@ def check_cntlr_physical_disks_len(cntrl_physical_disks):
+    boss_controller = [cntrl for cntrl in raid_controllers
+                       if drac_client.is_boss_controller(cntrl)]
+    if boss_controller:
+        return boss_controller[0]
```
@gaelrehault and @heydog15, what should be done if there is more than 1 BOSS found in the system? Is it OK to return the first one?

mpardhi23 replied:
@gaelrehault, confirmed that 2 BOSS cards is not valid configuration.

ajya commented on src/pilot/assign_role.py:
```diff
@@ -478,10 +485,10 @@ def define_storage_logical_disks(drac_client, raid_controllers):
     (os_logical_disk_size_gb,
-     os_physical_disk_names) = find_physical_disks_for_storage_os(
+     os_physical_disk_names) = find_physical_disks_for_os(
         physical_disks)
```
can try to wrap line 491 and 492 into one if it not as long anymore

mpardhi23 replied:
Done

ajya reviewed (changes requested):
thanks, looks good, some more comments added.

About testing - should test for case when there is only BOSS Controller. In stamps storage node can be used where there is BOSS controller and HBA330. As HBA330 is non-RAID controller, this system will be treated as having only BOSS controller.

mpardhi23 committed:
updated code for raid10 disk is_root_volume

gaelrehault commented on src/pilot/assign_role.py:
```diff
@@ -296,48 +296,116 @@ def get_raid_controller_ids(drac_client):
+        else:
+            LOG.critical("The BOSS card has only 1 SSD. "
+                         "2 SSDs are needed to configure a RAID 1")
+            return None
```
Correct, there should always be x2 disks available

mpardhi23 replied:
Thank you @gaelrehault  for this information!

mpardhi23 commented:
> thanks, looks good, some more comments added.
> 
> About testing - should test for case when there is only BOSS Controller. In stamps storage node can be used where there is BOSS controller and HBA330. As HBA330 is non-RAID controller, this system will be treated as having only BOSS controller.

Tested this code for the case when there is only BOSS controller , it is failing with this changes , need to add one condition  where it check controller should not be BOSS controller before below line. 
raid_10_logical_disk = define_single_raid_10_logical_disk(drac_client, raid_controller_ids[0], not boss_controller)

mpardhi23 committed:
added condition to fetch only raid controllers

mpardhi23 commented:
Hi @rpioso and @ajya,
I have updated latest changes which handles the warning part when more than one RAID controller and BOSS card found on server and selecting the first one from the list.

ajya reviewed (approved):
Thanks for updates

jproque-dell reviewed (approved):
Looks good to me. Nothing to add to what has been modified already.

mpardhi23 commented:
> Looks good to me. Nothing to add to what has been modified already.

Thank you JP for the review and approval !

rpioso reviewed (approved):
Looks great to me! Thank you, Monica.
